src/repeat.js

```diff
@@ -94,17 +115,74 @@ export class Repeat {
     }
   }
 
-  processItemsByStrategy() {
+  getInnerCollection() {
+    let expression = unwrapExpression(this.sourceExpression);
+    if (!expression) {
+      return null;
+    }
+    return expression.evaluate(this.scope, null);
+  }
+
+  observeInnerCollection() {
+    let items = this.getInnerCollection();
+    if (items instanceof Array) {
+      this.collectionObserver = this.observerLocator.getArrayObserver(items);
+    } else if (items instanceof Map) {
+      this.collectionObserver = this.observerLocator.getMapObserver(items);
+    } else {
+      return false;
+    }
+    this.callContext = 'handleInnerCollectionChanges';
+    this.collectionObserver.subscribe(this.callContext, this);
+    return true;
+  }
```

Comment:
This has not come up with respect to a repeater bug. Hopefully, we will have a repro soon. For reference, here's the discussion link: [November 10, 2015 10:13 PM](https://gitter.im/Aurelia/Discuss?at=5642b25bfa021fde389d03de)